lib/source_file.h

```diff
@@ -188,6 +189,10 @@ class SourceInfo final {
 
     const SourcePosition &getEnd() const { return this->end; }
 
+    /// Returns all the comments found in the file that is associated with this node.
+    /// Returns nothing if no sources are associated with the node.
+    [[nodiscard]] const std::vector<Comment *> getAllFileComments() const;
```

Comment:
Is this needed here? It is really weird to access all the comments in a file from *some* source info that points to a part of the file. Getting it from `InputSources` obtained by other means would make more sense to me if at all possible.

Reply:
Yes, this was kind of hackish.The only other way to access this was through `parserDriver`, so I'm fetching the comments now directly from it. Let me know if this looks alright.